File: codes/javascript/src/conditionals.js
/* eslint-disable eqeqeq */
/*
Relational Operators / Comparison Operators:
equal to (==) *checks for equality after performing type coercion. It converts operands to the same type before making the comparison.
not equal to (!=) *checks for inequality after performing type coercion. It converts operands to the same type before making the comparison.
strict equal to (===) *checks for strict equality without type coercion. It requires both the value and the type to be the same.
strict not equal to (!==) *checks for strict inequality without type coercion. It requires both the value and the type to be the same.
greater than (>)
less than (<)
greater then or equal to (>=)
less then or equal to (<=)

Logical Operators:
Logical AND (&&)
Logical OR (||)
Logical NOT (!)
*/

const CORRECT_ANSWER = 100;

let myAge;
let myAnswer;
let hasDrivingLicense;
let canDrive;

console.log("\n// equal to (==)\n");

myAnswer = "100";
console.log("myAnswer:", myAnswer);
console.log("CORRECT_ANSWER:", CORRECT_ANSWER);
console.log("(myAnswer == CORRECT_ANSWER):");
if (myAnswer == CORRECT_ANSWER) {
    console.log("Your answer is correct");
} else {
    console.log("Your answer is wrong");
}

console.log("\n// not equal to (!=)\n");

myAnswer = "25";
console.log("myAnswer:", myAnswer);
console.log("CORRECT_ANSWER:", CORRECT_ANSWER);
console.log("(myAnswer != CORRECT_ANSWER):");
if (myAnswer != CORRECT_ANSWER) {
    console.log("Your answer is correct");
} else {
    console.log("Your answer is wrong");
}

console.log("\n// strict equal to (===)\n");

myAnswer = "100";
console.log("myAnswer:", myAnswer);
console.log("CORRECT_ANSWER:", CORRECT_ANSWER);
console.log("(myAnswer === CORRECT_ANSWER):");
if (myAnswer === CORRECT_ANSWER) {
    console.log("Your answer is correct");
} else {
    console.log("Your answer is wrong");
}

console.log("\n// strict not equal to (!==)\n");

myAnswer = "25";
console.log("myAnswer:", myAnswer);
console.log("CORRECT_ANSWER:", CORRECT_ANSWER);
console.log("(myAnswer !== CORRECT_ANSWER):");
if (myAnswer !== CORRECT_ANSWER) {
    console.log("Your answer is correct");
} else {
    console.log("Your answer is wrong");
}

console.log("\n// greater than (>)\n");

myAge = 70;
console.log("myAge:", myAge);
console.log("(myAge > 50):");
if (myAge > 50) console.log("You are old");

console.log("\n// less than (<)\n");

myAge = 16;
console.log("myAge:", myAge);
console.log("(myAge < 20):");
if (myAge < 20) console.log("You are young");

console.log("\n// greater then or equal to (>=)\n");

myAge = 73;
console.log("myAge:", myAge);
console.log("(myAge >= 65):");
if (myAge >= 65) console.log("You are allowed to retire");

console.log("\n// less then or equal to (<=)\n");

myAge = 14;
console.log("myAge:", myAge);
console.log("(myAge <= 16):");
if (myAge <= 16) console.log("You are not allowed to drive");

console.log("\n// Logical AND (&&)\n");

myAge = 17;
console.log("myAge:", myAge);
hasDrivingLicense = false;
console.log("has driving license:", hasDrivingLicense);
console.log("((myAge >= 17) && (hasDrivingLicense === true)):");
console.log((((myAge >= 17) && (hasDrivingLicense === true)) ? "You are allowed to drive" : "You are not allowed to drive"));

myAge = 17;
console.log("myAge:", myAge);
hasDrivingLicense = true;
console.log("has driving license:", hasDrivingLicense);
console.log("((myAge >= 17) && (hasDrivingLicense === true)):");
console.log((((myAge >= 17) && (hasDrivingLicense === true)) ? "You are allowed to drive" : "You are not allowed to drive"));

console.log("\n// Logical OR (||)\n");

myAge = 2;
console.log("myAge:", myAge);
console.log("((myAge <= 3) || (myAge >= 65)):");
if ((myAge <= 3) || (myAge >= 65)) console.log("You should stay home");

myAge = 89;
console.log("myAge:", myAge);
console.log("((myAge <= 3) || (myAge >= 65)):");
if ((myAge <= 3) || (myAge >= 65)) console.log("You should stay home");

console.log("\n// Logical NOT (!)\n");

canDrive = false;
console.log("canDrive:", canDrive);
console.log("!canDrive:", !canDrive);

canDrive = true;
console.log("canDrive:", canDrive);
console.log("!canDrive:", !canDrive);

File: codes/ruby/conditionals.rb
=begin
Relational Operators / Comparison Operators:
equal to (==)
not equal to (!=)
greater than (>)
less than (<)
greater then or equal to (>=)
less then or equal to (<=)

Logical Operators:
Logical AND [&&, and]
Logical OR [||, or]
Logical NOT [!, not(x)]
=end

CORRECT_ANSWER = 100

print("\n# equal to (==)\n")

my_answer = 100.0
print("my_answer: ", my_answer, "\n")
print("CORRECT_ANSWER: ", CORRECT_ANSWER, "\n")
print("(my_answer == CORRECT_ANSWER): ")
if (my_answer == CORRECT_ANSWER)
  print("Your answer is correct")
else
  print("Your answer is wrong")
end

print("\n# not equal to (!=)\n")

my_answer = 25.0
print("my_answer: ", my_answer, "\n")
print("CORRECT_ANSWER: ", CORRECT_ANSWER, "\n")
print("(my_answer != CORRECT_ANSWER): ")
if (my_answer != CORRECT_ANSWER)
  print("Your answer is correct")
else
  print("Your answer is wrong")
end

print("\n# greater than (>)\n")

my_age = 70
print("my_age: ", my_age, "\n")
print("(my_age > 50): ")
print("You are old") if (my_age > 50)

print("\n# less than (<)\n")

my_age = 16
print("my_age: ", my_age, "\n")
print("(my_age < 20): ")
print("You are young") if (my_age < 20)

print("\n# greater then or equal to (>=)\n")

my_age = 73
print("my_age: ", my_age, "\n")
print("(my_age >= 65): ")
print("You are allowed to retire") if (my_age >= 65)

print("\n# less then or equal to (<=)\n")

my_age = 14
print("my_age: ", my_age, "\n")
print("(my_age <= 16): ")
print("You are not allowed to drive") if (my_age <= 16)

print("\n# Logical AND [&&, and]\n")

my_age = 17
print("my_age: ", my_age, "\n")
has_driving_license = false
print("has_driving_license: ", has_driving_license, "\n")
print("((my_age >= 17) && (has_driving_license == true)): ")
if ((my_age >= 17) && (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

my_age = 17
print("my_age: ", my_age, "\n")
has_driving_license = true
print("has_driving_license: ", has_driving_license, "\n")
print("((my_age >= 17) && (has_driving_license == true)): ")
if ((my_age >= 17) && (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

my_age = 17
print("my_age: ", my_age, "\n")
has_driving_license = false
print("has_driving_license: ", has_driving_license, "\n")
print("((my_age >= 17) and (has_driving_license == true)): ")
if ((my_age >= 17) and (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

my_age = 17
print("my_age: ", my_age, "\n")
has_driving_license = true
print("has_driving_license: ", has_driving_license, "\n")
print("((my_age >= 17) and (has_driving_license == true)): ")
if ((my_age >= 17) and (has_driving_license == true)) then print("You are allowed to drive") else print("You are not allowed to drive") end

print("\n# Logical OR [||, or]\n")

my_age = 2
print("my_age: ", my_age, "\n")
print("((my_age <= 3) || (my_age >= 65)): ")
print("You should stay home") if ((my_age <= 3) || (my_age >= 65))

my_age = 89
print("my_age: ", my_age, "\n")
print("((my_age <= 3) || (my_age >= 65)): ")
print("You should stay home") if ((my_age <= 3) || (my_age >= 65))

my_age = 2
print("my_age: ", my_age, "\n")
print("((my_age <= 3) or (my_age >= 65)): ")
print("You should stay home") if ((my_age <= 3) or (my_age >= 65))

my_age = 89
print("my_age: ", my_age, "\n")
print("((my_age <= 3) or (my_age >= 65)): ")
print("You should stay home") if ((my_age <= 3) or (my_age >= 65))

print("\n# Logical NOT [!, not]\n")

can_drive = false
print("can_drive:", can_drive, "\n")
print("!can_drive:", !can_drive, "\n")

can_drive = true
print("can_drive:", can_drive, "\n")
print("!can_drive:", !can_drive, "\n")

can_drive = false
print("can_drive:", can_drive, "\n")
print("not can_drive:", not(can_drive), "\n")

can_drive = true
print("can_drive:", can_drive, "\n")
print("not can_drive:", not(can_drive), "\n")
